src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/ClusterLocalRegistrar.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    /// <summary>
+    /// The registrar for the Cluster-Local Registration Strategy.
+    /// </summary>
+    internal class ClusterLocalRegistrar : IGrainRegistrar
```

Comment:
I agree with @gabikliot here. If the implementation is sync today, hiding it behind an async API is potentially misleading. Maybe ClusterLocalRegistrar shouldn't implement IGrainRegistrar to begin with? That way it'll always be obvious on the caller side if the call is local/sync vs. remote/async.